docs/reference/technology/connectivity/networking.md

```diff
@@ -0,0 +1,119 @@
+### IP Networking: WORK-IN-PROGRESS
+
+IP Networking in Mbed OS is layered in three clearly defined API levels. In the diagram below,
+the layers are shown next to closest matching [OSI model](https://en.wikipedia.org/wiki/OSI_model) layers.
+
+![ip-networking](ip-networking.png)
+
+Following sections introduce the APIs and technologies implemented in each level.
+
+#### Socket API
+
+Socket API is the common API between all IP connectivity methods. All network stacks in Mbed OS
+provide the same Socket API, making applications portable between different connectivity methods
+or even stacks.
+
+In OSI model, Socket API relates to layer 4, Transport layer. In Mbed OS Socket API supports
+both [TCP](https://en.wikipedia.org/wiki/Transmission_Control_Protocol) and [UDP](https://en.wikipedia.org/wiki/User_Datagram_Protocol) protocols.
+
+Refer to [TODO:Socket API](networksockets.md) reference for usage instructions.
+
+#### IP Stacks
+
+Mbed OS has three options to select for IP stack. Two build-in IP stack or external IP
+stack provided by the connectivity module.
+
+![Network stack](networkstacks.png)
+
+As shown from the diagram above, all stacks implement the same Socket API and
+therefore application developer rarely needs to know which stack is going to be used.
+Mbed OS chooses one at the build time and usually it is LwIP.
+
+Some external WiFi modules and most cellular modules are in fact external IP stacks from
+application point of view. In that case the network driver implements actually the full Network
+stack API. These drivers are usually driving the module through AT-command type of interface.
+It saves RAM and Flash to use external IP module but depending on its driver and AT-command
+interface, it might not properly match the full Socket API.
+
+Following table summarises different stacks, usecases and their limitations.
+
+|Stack|Network protocols supported|Use cases|Limitations|
+|-----|---------------------------|---------|-----------|
+|LwIP|IPv4, IPv6, PPP|Ethernet, WiFi, 2G/3G/4G Cellular|4 sockets, 1 interface, no routing|
```

Comment:
4 sockets is configuration option, so I think it is not a limitation. Application designer can choose to configure more sockets to lwip, with a cost of additional ram.

Reply:
I know it can be configured. Many things can be, but to keep this page simple, I don't wan't to clutter it with extra information that will be presented later in the same book.

Technology section is before configuration section, so I don't want to refer it before it has been introduced.